test/unit/eval_algorithms/test_factual_knowledge.py

```diff
@@ -40,10 +40,48 @@ def test_factual_knowledge_invalid_config(self):
         with pytest.raises(EvalAlgorithmClientError, match=re.escape(expected_error_message)):
             FactualKnowledgeConfig(target_output_delimiter="")
 
+    def test_factual_knowledge_invalid_config_logical_operator(self):
+        """
+        GIVEN invalid inputs for logical_operator
+        WHEN FactualKnowledgeConfig is initialized
+        THEN correct exception with proper message is raised
+        """
+        logical_operator_error_message = (
+            'Invalid logical_operator is provided. The only valid inputs are strings "OR" and "AND".'
+        )
+        invalid_logical_operators = ["", "NOT AND", "Random", "<OR>"]
+        for invalid_input in invalid_logical_operators:
```

Comment:
Instead of using for loops, please parametrize the test arguments with `@pytest.mark.parametrize`

Reply:
You can modify the existing `test_factual_knowledge_invalid_config` in line 31, for the argument you can specify different error message, logical_operator and target_delimiter.